### ISSUE TITLE
Binaries for v1.13.0 were changed after the release was created, causing a hash mishmatch

### ISSUE THREAD

**Issue by birjj** [Azure/terraform-provider-azapi#477]
<small>_Duplicate of #473 and #461, but since those issues are closed without the actual problem being solved, I figured I'd open another with an actual explanation of why the problem happens._</small>

When trying to install the newest version of provider using OpenTofu, the following error happens:

```
Error while installing azure/azapi v1.13.0: checksum list has unexpected
SHA-256 hash 18e1464f3e11436f04f3fabd842a079d47a2d7b56489e8d920d9cd5156bbd52e
(expected 8170881d34e63fdade2a382c6b3d0bbbd68ffc0a1527835b4fc113cd8726acc6)
```

This is caused by a mismatch between the expected hash, and the actual hash for the `azapi` provider binary (in this case of the `terraform-provider-azapi_1.13.0_linux_amd64.zip` file).

## Analysis

The OpenTofu registry scrapes new provider versions frequently, and stores the results in their GitHub repo. In the case of the `azapi` provider, that file is [here](https://github.com/opentofu/registry/blob/d0b924dc954dd5a1946cc13b6035c3df001b12ce/providers/a/Azure/azapi.json#L65), with the newest v1.13.0 changes:

```
{
    "os": "linux",
    "arch": "amd64",
    "filename": "terraform-provider-azapi_1.13.0_linux_amd64.zip",
    "download_url": "https://github.com/Azure/terraform-provider-azapi/releases/download/v1.13.0/terraform-provider-azapi_1.13.0_linux_amd64.zip",
    "shasum": "8170881d34e63fdade2a382c6b3d0bbbd68ffc0a1527835b4fc113cd8726acc6"
},
```

These contain the expected hashes (`817....` from the error), which don't match the current hashes (`18e...`) of the v1.13.0 release.  
OpenTofu scraped those [in this GitHub Actions run](https://github.com/opentofu/registry/actions/runs/8815278498/job/24196908305#step:4:6923) at 2024-04-24T10:47:25Z, which it did by downloading the `_SHA256SUMS` file from the v1.13.0 release from `terraform-prodivder-azapi`. At that time, that file contained the old hashes (`817...`).

At some point later, the v1.13.0 release of `terraform-provider-azapi` must have been updated with new files. However, since the version wasn't bumped, this didn't propagate into the OpenTofu repository.

I haven't been able to find out how `terraform-provider-azapi` creates its releases.

**TL;DR: an earlier release of `terraform-provider-azapi` v1.13.0 contained the old hashes (`817...`). At some point later, the release must have been updated in-place, without creating a new version. This causes the hash mismatch.**

It obviously seems fishy that the binaries were changed after the release was made. The commit that the v1.13.0 tag points to was created at 2024-04-24T05:05:10Z (several hours before OpenTofu scraped the release), so whatever changes were made _after_ OpenTofu scraped the release isn't documented in the git history.

**Comment by birjj** [comment]
Thanks for the quick response @ms-henglu, and for clarifying why the hashes were updated. That makes sense (and [happens to the best of us :wink:](https://discuss.hashicorp.com/t/change-in-sha256sums-for-terraform-downloads/52522/2)).

**Comment by ms-henglu** [comment]
Hi @birjj ,

The bug fix has been released in v1.13.1, would you please have a try? Thanks.

**Comment by meysam81** [comment]
@ms-henglu 
Is this the desired behavior?

```
- Installed azure/azapi v1.13.1. Signature validation was skipped due to the registry not containing GPG keys for this provider
```

**Comment by ms-henglu** [comment]
Hi @meysam81 ,

I believe you're using `opentofu`, I found a related issue about it: https://github.com/opentofu/opentofu/issues/266

We haven't shared the public key which is used to verify the signature to opentofu.

**Comment by birjj** [comment]
I can confirm that the hashes match for v1.13.1 in the OpenTofu registry

**Comment by evotch** [comment]
I get the same error with 1.13.1 

Error while installing azure/azapi v1.13.1: archive has incorrect checksum zh:1747a2e3b341f2c05a3a54c6998a95ba00284db0d3d21767c5ba10fcff8505c2 (expected zh:e3eb48ee8c36ee4f81850d8a21fc59b81886c729d7c3b7adece4a25f355bed2f)

**Comment by birjj** [comment]
@evotch I'm assuming you're on macOS with an Apple chip, based on your expected hash. (`e3e...` which matches the `terraform-provider-azapi_1.13.1_darwin_arm64.zip` file).

The hash you're getting (`174...`) is not the computed checksum of the `terraform-provider-azapi_1.13.1_darwin_arm64.zip` file from [the release](https://github.com/Azure/terraform-provider-azapi/releases/tag/v1.13.1). When I download that file and verify its sha256 checksum I correctly get `e3e...`. It therefore doesn't seem to be the same problem as described in this issue.

Perhaps the file failed to download completely?
